### PR TITLE
Improves message when transitively serializing PipelineOptions

### PR DESCRIPTION
**Pull request by jkff**
R: @lukecwik

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java [hunk 1]
```diff
@@ -164,6 +166,16 @@ public Object invoke(Object proxy, Method method, Object[] args) {
         + Arrays.toString(args) + "].");
   }
 
+  private void writeObject(java.io.ObjectOutputStream stream)
+      throws IOException {
+    throw new NotSerializableException(
+        "PipelineOptions objects are not serializable and should not be embedded into transforms "
+            + "(did you capture a PipelineOptions object in a field or in an anonymous class?). "
+            + "Instead, if you're using a DoFn, access PipelineOptions at runtime "
+            + "from ProcessContext.pipelineOptions(), or pre-extract necessary fields from "
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java]: `ProcessContext.pipelineOptions()` -> `ProcessContext.getPipelineOptions()`

Its also available on StartBundleContext and FinishBundleContext, should we mention these?

**Comment by lukecwik** [comment]
Findbugs is complaining:
2017-06-12T18:58:47.767 [INFO] --- findbugs-maven-plugin:3.0.4:check (default) @ beam-sdks-java-core ---
2017-06-12T18:58:47.830 [INFO] BugInstance size is 3
2017-06-12T18:58:47.838 [INFO] Error size is 0
2017-06-12T18:58:47.885 [INFO] Total bugs: 3
2017-06-12T18:58:47.892 [INFO] Class org.apache.beam.sdk.options.ProxyInvocationHandler defines non-transient non-serializable instance field interfaceToProxyCache [org.apache.beam.sdk.options.ProxyInvocationHandler] In ProxyInvocationHandler.java SE_BAD_FIELD
2017-06-12T18:58:47.893 [INFO] Class org.apache.beam.sdk.options.ProxyInvocationHandler defines non-transient non-serializable instance field jsonOptions [org.apache.beam.sdk.options.ProxyInvocationHandler] In ProxyInvocationHandler.java SE_BAD_FIELD
2017-06-12T18:58:47.893 [INFO] Class org.apache.beam.sdk.options.ProxyInvocationHandler defines non-transient non-serializable instance field options [org.apache.beam.sdk.options.ProxyInvocationHandler] In ProxyInvocationHandler.java SE_BAD_FIELD

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java (hunk 1 above)
Comment (jkff) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java]: Mentioned now.

**Comment by jkff** [comment]
Fixed findbugs too, thanks (by adding an exclusion in findbugs-filter.xml).

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java [hunk 2]
```diff
@@ -164,6 +166,17 @@ public Object invoke(Object proxy, Method method, Object[] args) {
         + Arrays.toString(args) + "].");
   }
 
+  private void writeObject(java.io.ObjectOutputStream stream)
+      throws IOException {
+    throw new NotSerializableException(
+        "PipelineOptions objects are not serializable and should not be embedded into transforms "
+            + "(did you capture a PipelineOptions object in a field or in an anonymous class?). "
+            + "Instead, if you're using a DoFn, access PipelineOptions at runtime "
+            + "via ProcessContext/StartBundleContext/FinishBundleContext.pipelineOptions(), "
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java]: `.pipelineOptions` -> `.getPipelineOptions()`

**Pull request closed without merging**